File: backend/src/main/java/com/fdsa/altairis/service/CitiesService.java
package com.fdsa.altairis.service;

import com.fdsa.altairis.dto.request.CityRequest;
import com.fdsa.altairis.dto.response.CityResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CitiesService {

    CityResponse create(CityRequest request, String user);
    CityResponse update(Long id, CityRequest request, String user);
    CityResponse delete(Long id, String user);
    CityResponse getById(Long id);
    Page<CityResponse> list(Long countryId, Pageable pageable);

}

File: frontend/src/app/services/services/cities.service.ts
import { Injectable } from '@angular/core';
import { ApiClient } from '../../core/api/api.client';
import { PageResponse } from '../../core/api/api.types';
import { CityCreateRequest, CityUpdateRequest } from '../../interfaces/request/city.request';
import { CityResponse } from '../../interfaces/response/city.response';

@Injectable({ providedIn: 'root' })
export class CitiesService {
  constructor(private api: ApiClient) {}

  list(countryId: number | null, page: number, size: number) {
    return this.api.get<PageResponse<CityResponse>>('/api/cities', { countryId: countryId ?? undefined, page, size });
  }

  getById(id: number) {
    return this.api.get<CityResponse>(`/api/cities/${id}`);
  }

  create(req: CityCreateRequest) {
    return this.api.post<CityResponse>('/api/cities', req);
  }

  update(id: number, req: CityUpdateRequest) {
    return this.api.put<CityResponse>(`/api/cities/${id}`, req);
  }

  delete(id: number) {
    return this.api.delete(`/api/cities/${id}`);
  }
}
